src/realm/group_writer.cpp

```diff
@@ -485,8 +485,10 @@ ref_type GroupWriter::write_group()
 size_t GroupWriter::get_free_space() {
     if (m_free_lengths.is_attached()) {
         size_t sum = 0;
-        for (size_t j=0; j<m_free_lengths.size(); ++j)
-            sum += m_free_lengths.get(j);
+        for (size_t j = 0; j < m_free_lengths.size(); ++j) {
+            int64_t l = to_size_t(m_free_lengths.get(j));
```

Comment:
The line introduced here defeats the purpose of the call to `to_size_t` and the next line then again involves the two types that probably gave you the warning in the first place. 😛 Please revert these lines and only add `to_size_t()` to get the minimal diff.